### PR TITLE
Remove deprecated code for Sphinx 5.0

### PR DESCRIPTION
**Pull request by AA-Turner**
Remove all code marked with `RemovedInSphinx50Warning`, ahead of Sphinx 5.0's release.

A

### Feature or Bugfix
- Refactoring

### CODE REVIEW

File: sphinx/deprecation.py
Comment (tk0miya) [inline sphinx/deprecation.py]: Note: We need to change this to a child of `DeprecationWarning`
```suggestion
class RemovedInSphinx60Warning(DeprecationWarning):
```

File: sphinx/ext/autodoc/__init__.py
Comment (tk0miya) [inline sphinx/ext/autodoc/__init__.py:2853]: `"alphabetic"` should also be removed from the "add_config_value" call.

Comment (tk0miya) [reply sphinx/ext/autodoc/__init__.py:2853]: I'll do this in another PR after the merge.

Comment (AA-Turner) [reply sphinx/ext/autodoc/__init__.py:2853]: Done

Comment (tk0miya) [reply sphinx/ext/autodoc/__init__.py:2853]: Wow!

**Review by tk0miya — APPROVED** [review]
What a quick work! I'll merge this after CI passed.

**Comment by AA-Turner** [comment]
CI has passed!

**Comment by tk0miya** [comment]
Merged. Thank you for your work. This PR must be helpful because I've perfectly forgotten removing this.

**Pull request merged**
